woocommerce/woocommerce-ios
# Order Split View: Show the first item in the secondary column if possible

Part of #6381 


### Description

There are some existing issues in the order split view regarding the secondary column:
- When the list is first loaded, nothing is displayed on the secondary column even when the list on the primary column is not empty. It'd be nice to automatically display the first item on the secondary column.
- When applying filters to the list, if there was an item displaying on the secondary column, the column still displays the item even when it no longer exists on the list. It'd be nice to switch to the first item on the updated list or show an empty view if the list is empty.

This PR fixes the above issues with the following changes:
- Add a method to check the currently selected item in the list:
  - If no item is selected yet, attempt to highlight the first item in the list and display it on the secondary column.
  - If there's a selected item but no item can be found at the selected index path, switch to displaying the first item instead.
  - If there's an item at the selected index path, but its ID doesn't match the selected order ID, also switch to displaying the first item instead.
- Update the secondary column to display an empty view if we cannot find an appropriate order to display. 


### Testing instructions

- Run the app in an iPad or landscape mode of a large-screened iPhone.
- Navigate to Orders tab, notice that the first item in the order list is automatically displayed on the secondary column.
- Apply some filters to the list, notice that the secondary column is updated to show the first item in the list. If the list is empty, the secondary column displays an empty view.
- Please feel free to run the app with the feature flag `.splitViewInOrdersTab` disabled to make sure that everything still works as before.

### Screenshots


https://user-images.githubusercontent.com/5533851/170672356-a9c719f5-dbf1-4004-b571-d5e95011e307.mp4



---
